api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 package v1beta1
 
+// AzureVMImageType is used to specify the source of the Azure VM boot image.
+// Valid values are ImageID and AzureMarketplace.
+// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
 type AzureVMImageType string
 
 const (
-	ImageID          AzureVMImageType = "ImageID"
+	// ImageID is the used to specify that an Azure resource ID of a VHD image is used to boot the Azure VMs from.
+	ImageID AzureVMImageType = "ImageID"
+
+	// AzureMarketplace is used to specify the Azure Marketplace image info to use to boot the Azure VMs from.
 	AzureMarketplace AzureVMImageType = "AzureMarketplace"
 )
 
+// AzureNodePoolPlatform is the platform specific configuration for an Azure node pool.
 type AzureNodePoolPlatform struct {
-	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
 	//
+	// TODO: This should be validated with a regex, based on https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
 	// +kubebuilder:validation:Required
-	VMSize string `json:"vmsize"`
+	VMSize string `json:"vmSize"`
 
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used and
+	// image is used to configure the VM boot image. If unset, the default image at the location below will be used and
 	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
 	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
 	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
 	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
 	//
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Image is immutable"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
 	// +kubebuilder:validation:Required
 	Image AzureVMImage `json:"image"`
 
-	// DiskSizeGB is the size in GB to assign to the OS disk
-	// CAPZ default is 30GB, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b3708019a67ff19407b87d63c402af94ca4246f6/api/v1beta1/types.go#L599
+	// diskSizeGB is the size in GB to assign to the OS disk. This should be between 16 and <TODO find a limit>.
```

Comment:
Should set a limit, even if it is arbitrarily high

Reply:
100 TiB? per https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk-size

Reply:
I think the limits are set out here, https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types, the 100TiB appears to be across your account, not for an individual disk

Reply:
Ah true. 65,536 then?

Reply:
I think so, and we can use CEL to limit it for the smaller types too, I'll push up something to show this